staging/src/k8s.io/cloud-provider/plugins.go

```diff
@@ -91,6 +91,23 @@ func IsExternal(name string) bool {
 	return name == externalCloudProvider
 }
 
+// IsDeprecatedInternal is responsible for preventing cloud.Interface
+// from being initialized in kubelet, kube-controller-manager or kube-api-server
+func IsDeprecatedInternal(name string) bool {
+	for _, provider := range deprecatedCloudProviders {
+		if provider.name == name {
+			klog.Infof("INFO: Please make sure you are running external cloud controller manager binary for provider %q."+
```

Comment:
I think we should remove these log lines in `IsDeprecatedInternal` and just improve the warnings in `DeprecationWarningForProvider` and ensure `DeprecationWarningForProvider` is called whenever `IsDeprecatedInternal` is called. What do you think?

Reply:
We can't merge any of this. `IsDeprecatedInternal` is talking about disabled cloud providers, and `DeprecationWarningForProvider` should not ever be called with it. It is talking about deprecated stage of the cloud provider, which already passed when IsDeprecatedInternal returns true. The logs should never overlap. In both occasions detailed infro for cloud provider is already updated with pointers to the repository.

Reply:
yeah my point is to just separate logging and the "deprecation" check in two different functions. There are already cases we call both, so we're not logging twice

Reply:
(the log messages here are jsut copied from DeprecationWarningForProvider)

Reply:
Added this in a separate method. Marking as resolved.